payroll/models/hr_payslip.py

```diff
@@ -587,117 +468,135 @@ def get_contract_dict(self, contract, contracts):
 
         return {}
 
-    @api.model
-    def _get_payslip_lines(self, contract_ids, payslip_id):
-        def _sum_salary_rule_category(localdict, category, amount):
-            if category.parent_id:
-                localdict = _sum_salary_rule_category(
-                    localdict, category.parent_id, amount
-                )
-
-            if category.code in localdict["categories"].dict:
-                localdict["categories"].dict[category.code] += amount
-            else:
-                localdict["categories"].dict[category.code] = amount
-
-            return localdict
-
-        payslip = self.env["hr.payslip"].browse(payslip_id)
-        contracts = self.env["hr.contract"].browse(contract_ids)
-
-        # we keep a dict with the result because a value can be overwritten by
-        # another rule with the same code
-        result_dict = {}
-        rules_dict = {}
-        contract_dict = {}
-        blacklist = []
-        categories = BrowsableObject(payslip.employee_id.id, {}, self.env)
-        rules = BrowsableObject(payslip.employee_id.id, rules_dict, self.env)
-        current_contract = BrowsableObject(
-            payslip.employee_id.id, contract_dict, self.env
-        )
-
-        baselocaldict = payslip.get_baselocaldict(contracts)
-        baselocaldict["categories"] = categories
-        baselocaldict["rules"] = rules
-        baselocaldict["current_contract"] = current_contract
+    def _get_baselocaldict(self, contracts):
+        self.ensure_one()
+        worked_days_dict = {
+            line.code: line for line in self.worked_days_line_ids if line.code
+        }
+        input_lines_dict = {
+            line.code: line for line in self.input_line_ids if line.code
+        }
+        localdict = {
+            **{
```

Comment:
If you currently don't need this I would prefer you leave it out. If you use it in the future it can always be re-introduced.

Reply:
Actually I forgot to tell you. I texted removing the ** and if I do then I get an error because a dict is not accesible. 

I think I put that there because of that. Take into account that the dict is inside another dict and that's why ** allow merging it with the parent dict and make the values accessible. 

I can try to remove it completely and only leave one dict but en last commit I tested without it and all tests fail. 

Will try getting it out in a while and if not we will have to leave it there.